src/core/exception_hacks.cc

```diff
@@ -168,7 +168,10 @@ int dl_iterate_phdr(int (*callback) (struct dl_phdr_info *info, size_t size, voi
     return r;
 }
 
-#ifndef NO_EXCEPTION_INTERCEPT
+// We disable interception of _Unwind_RaiseException when ASAN is enabled
+// since it can produce a large number of stack-related false positives when
+// it is enabled and exceptions are thrown.
+#if !defined(NO_EXCEPTION_INTERCEPT) && !defined(SEASTAR_ASAN_ENABLED)
```

Comment:
Any reason we can't just set `NO_EXCEPTION_INTERCEPT` on our side when enabling ASAN?

Reply:
This also works - is there some reason to prefer it?

I went this way since it seems like you always want this (i.e., there is no point in exposing it as a degree of freedom in the build system, rather the build system would just always set this when asan is enabled) and here in the code just works exactly when ASAN is enabled regardless of build system (since for example we still use cmake in CI, etc).

Is your concern the burden of upstreaming it?

Reply:
> Is your concern the burden of upstreaming it?

Coming from that angle yes. I mean I was kinda thinking you could just do that now then you don't have to deal with the fork business until it's upstreamed. But the PR is up now so 🤷

Reply:
I guess I'll leave it then. I seems reasonable that it could be accepted in its current form upstream.